src/syntax/ast.rs

```diff
@@ -5,101 +5,122 @@ pub struct Resource {
 }
 
 #[derive(Debug, PartialEq)]
-pub struct Comment {
-    pub body: String,
-}
-
-#[derive(Debug, PartialEq)]
-pub struct Section {
-    pub key: Key,
-    pub body: Vec<Entry>,
+pub struct Message {
+    pub id: Identifier,
+    pub value: Option<Pattern>,
+    pub attributes: Option<Vec<Attribute>>,
     pub comment: Option<Comment>,
 }
 
 #[derive(Debug, PartialEq)]
 pub enum Entry {
     Message(Message),
-    Comment(Comment),
     Section(Section),
+    Comment(Comment),
     Junk(JunkEntry),
 }
 
 #[derive(Debug, PartialEq)]
-pub struct Message {
+pub struct Pattern {
+    pub elements: Vec<PatternElement>,
+    pub quoted: bool,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum Expression {
+    Selector(SelectorExpression),
+    SelectExpression {
+        exp: Option<SelectorExpression>,
+        variants: Vec<Variant>
+    },
+}
+
+#[derive(Debug, PartialEq)]
+pub enum SelectorExpression {
+    Pattern(Pattern),
+    MessageReference(String),
+    ExternalArgument(String),
+    CallExpression {
+        callee: Builtin,
+        args: Vec<SelectorExpression>,
```

Comment:
If we leave this as-is, would you agree to the following renames?

- `Expression` → `PlaceableExpression`,
- `SelectorExpression` → `Expression`